Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4345,6 +4345,9 @@ void WebPageProxy::receivedNavigationActionPolicyDecision(WebProcessProxy& proce
             }
         }
 
+        if (auto* websitePolicies = navigation->websitePolicies(); websitePolicies && frame->isMainFrame())
+            m_mainFrameWebsitePoliciesData = websitePolicies->data();
```

Comment:
This is not the correct time to set this.  If you're on a.com and navigating to b.com, there is still a lot of time between this point and when the navigation is committed when we can make an iframe, and that iframe would have b.com's policies.  It looks like you already have the right code in WebPageProxy::commitProvisionalPage, so this should be removed.